sparse/_common.py

```diff
@@ -1212,6 +1212,173 @@ def _dot_ndarray_coo(array1, coords2, data2, out_shape):  # pragma: no cover
     return _dot_ndarray_coo
 
 
+def _parse_einsum_input(operands):
```

Comment:
Please leave a comment where this was taken and a link to the airport license.

Reply:
Sure, i have add this:

`Copied from : https://github.com/numpy/numpy/blob/main/numpy/core/einsumfunc.py
    under BSD-3-Clause license : https://github.com/numpy/numpy/blob/main/LICENSE.txt`

Do you think it is sufficient? (Licence are the same in sparse and numpy so that shouldn't be an issue and BSD-3 allow for copy and modification)

I have also removed the part of the code that was causing the fail of the test

Reply:
I'd suggest adding it in a comment above the function instead of in the docstring, otherwise the note looks fine. I'd also link the exact commit, rather than Main which can have changing line numbers.

Reply:
Ok, done